pkg/utils/gardener/secretsrotation/etcd.go

```diff
@@ -165,10 +168,14 @@ func rewriteEncryptedData(
 
 				// Wait until we are allowed by the limiter to not overload the API server with too many requests.
 				if err := limiter.Wait(ctx); err != nil {
-					return err
+					return fmt.Errorf("rate limiter wait failed: %w", err)
+				}
+
+				if err := c.Patch(ctx, &obj, patch); err != nil {
+					return fmt.Errorf("failed to patch object when rewriting encrypted data: %w", err)
```

Comment:
I suggest to omit the `when rewriting encrypted data` part. It will be added by the caller.

The final error would look like: 

```
failed to rewrite encrypted data: failed to patch object: <wrapped error>
```

instead of

```
failed to rewrite encrypted data: failed to patch object when rewriting encrypted data: <wrapped error>
```

```suggestion
					return fmt.Errorf("failed to patch object: %w", err)
```